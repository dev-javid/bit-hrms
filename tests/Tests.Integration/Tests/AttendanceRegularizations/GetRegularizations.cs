namespace Tests.Integration.Tests.AttendanceRegularizations
{
    public class GetRegularizations(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/attendance-regularizations";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data_Of_Other_Employee_For_Company_Admin()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/AttendanceRegularizations/GetRegularizations.sql");

            using var response = await Client.GetAsync($"{Route}?EmployeeId=2");

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Own_Employee_Data_For_Employee()
        {
            await LoginAsEmployeeAsync();
            await FeedDataAsync("Tests/AttendanceRegularizations/GetRegularizations.sql");

            using var response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
