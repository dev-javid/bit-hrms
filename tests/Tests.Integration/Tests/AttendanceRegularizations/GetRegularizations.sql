INSERT INTO public. attendance_regularization
(id, employee_id, date, clock_in_time, clock_out_time, reason, approved, created_by, created_on, modified_by, modified_on, company_id)
VALUES
(3, 1, '2024-05-03', '08:00:00', '14:00:00', 'sick', true, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL, 999),
(4, 1, '2024-06-04', '08:00:00', '15:00:00', 'sick', true, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL, 999),
(5, 1, '2024-06-05', '08:00:00', '16:00:00', 'sick', true, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL, 999),
(6, 1, '2024-07-06', '08:00:00', '14:00:00', 'sick', true, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL, 999),
(7, 2, '2024-07-07', '08:00:00', '14:00:00', 'sick', true, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL, 999),
(8, 2, '2024-07-08', '08:00:00', '18:00:00', 'sick', true, 0, '2024-05-03 03:31:32.227954-07', NULL, NULL, 999);

