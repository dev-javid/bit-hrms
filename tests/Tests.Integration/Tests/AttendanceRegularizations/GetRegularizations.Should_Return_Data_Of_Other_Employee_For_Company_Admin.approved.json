{
    "page": 1,
    "limit": 10,
    "total": 2,
    "items": [
        {
            "attendanceRegularizationId": 8,
            "employeeId": 2,
            "date": "2024-07-08",
            "clockInTime": "08:00:00",
            "clockOutTime": "18:00:00",
            "reason": "sick",
            "approved": true
        },
        {
            "attendanceRegularizationId": 7,
            "employeeId": 2,
            "date": "2024-07-07",
            "clockInTime": "08:00:00",
            "clockOutTime": "14:00:00",
            "reason": "sick",
            "approved": true
        }
    ]
}